### ISSUE TITLE
Analyzer does not iterate dependencies of build.gradle.kts for Gradle packages

### ISSUE THREAD

**Issue by schvvarzekatze** [oss-review-toolkit/ort#9231]
### When using

For Gradle packages an analyzer result without analyze of relevant dependencies is generated. This leads to an empty license NOTICE file.

The same config including environment is only working and generation successful reports for Yarn as package manager. 

For an older ORT image (about 2 years) analyzer result is also sucessfully generated for Gradle. (recent workaround to generate reports for Yarn and Gradle.

Question: Are there special config changes to be considered to make a recent ORT version run as well for Gradle?

I checked the manual but did not find relevant config changes already solving my issue.

### To Reproduce

1. Create flattened project with all dependencies
2. Start ORT analyzer for Package Manager Gradle

### Expected behavior

Libraries listed in `build.gradle.kts` are analyzed

### Console / log output

Logs

```
/        \_______   \__    ___/ The OSS Review Toolkit, version DOCKER-SNAPSHOT
|    |   | |       _/ |    |    built with JDK 11.0.24+8, running under Java 17
|    |   | |    |   \ |    |    Executing 'analyze' as 'root' on Linux         
\________/ |____|___/ |____|    with 8 CPUs and a maximum of 19664 MiB of memory
                                                                               
Environment variables:                                                        
ORT_CONFIG_DIR = /xyz-project/ort                                             
ORT_DATA_DIR = /root/.ort                                                     
HOME = /home/ort                                                              
JAVA_HOME = /opt/java/openjdk                                                 
ANDROID_HOME = /opt/android-sdk                                               
                                                                              
Looking for ORT configuration in the following file:
        /xyz-project/ort/config.yml

Looking for analyzer-specific configuration in the following files and directories:
        /project/.ort.yml
        /xyz-project/ort/resolutions.yml
The following 1 package manager(s) are enabled:
        Gradle
The following 3 package curation provider(s) are enabled:
        RepositoryConfiguration, DefaultDir, DefaultFile
Analyzing project path:
        /project
15:02:58.238 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/project/src/main/resources' as it is hard-coded to be ignored.
15:02:59.176 [main] INFO  org.ossreviewtoolkit.utils.ort.OrtAuthenticator - Authenticator was successfully installed.
15:02:59.194 [main] INFO  org.ossreviewtoolkit.utils.ort.OrtProxySelector - Proxy selector was successfully installed.
Found 1 Gradle definition file(s) at:
        build.gradle.kts
Found in total 1 definition file(s) from the following 1 package manager(s):
        Gradle
15:02:59.259 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManagerRunner - Starting Gradle analysis.
15:02:59.262 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Using Gradle to resolve dependencies for path 'build.gradle.kts'...
Downloading https://services.gradle.org/distributions/gradle-8.10-bin.zip
..................................................................................................................................
15:03:11.586 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.plugins.packagemanagers.gradle.Gradle - The project at '/project' uses Gradle version 8.10.0.
15:03:41.470 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'hg root' in '/project'...
15:03:41.823 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Gradle resolved dependencies for path 'build.gradle.kts' in 42.557892008s.
15:03:41.827 [DefaultDispatcher-worker-1] INFO  kotlinx.coroutines.CoroutineScope - Finished Gradle analysis.
15:03:41.848 [main] INFO  org.ossreviewtoolkit.utils.config.ConfigurationResolver - Getting 0 package curation(s) from provider 'RepositoryConfiguration' took 262.379us.
15:03:41.848 [main] INFO  org.ossreviewtoolkit.utils.config.ConfigurationResolver - Getting 0 package curation(s) from provider 'DefaultDir' took 3.251us.
15:03:41.849 [main] INFO  org.ossreviewtoolkit.utils.config.ConfigurationResolver - Getting 0 package curation(s) from provider 'DefaultFile' took 290.011us.
Wrote analyzer result to '/xyz-project/build/oss-review-toolkit/gradle/analyzer/analyzer-result.yml' (0.01 MiB) in 595.436530ms.
The analysis took 42.636105019s.
Found 1 project(s) and 0 package(s) in total (not counting excluded ones).
Applied 0 curation(s) from 0 of 3 provider(s).
Resolved issues: 0 errors, 0 warnings, 0 hints.
Unresolved issues: 0 errors, 0 warnings, 0 hints.

```

Analyzer result
```
analyzer:
  start_time: "2024-10-01T15:02:59.202420646Z"
  end_time: "2024-10-01T15:03:41.838525665Z"
  environment:
    ort_version: "DOCKER-SNAPSHOT"
    build_jdk: "11.0.24+8"
    java_version: "17.0.5"
    os: "Linux"
    processors: 8
    max_memory: 20619198464
    variables:
      HOME: "/home/ort"
      JAVA_HOME: "/opt/java/openjdk"
      ANDROID_HOME: "/opt/android-sdk"
    tool_versions: {}
  config:
    allow_dynamic_versions: true
    enabled_package_managers:
    - "gradle"
    skip_excluded: false
  result:
    projects:
    - id: "Gradle::flattened-xyz:"
      definition_file_path: ""
      declared_licenses: []
      declared_licenses_processed: {}
      vcs:
        type: ""
        url: ""
        revision: ""
        path: ""
      vcs_processed:
        type: ""
        url: ""
        revision: ""
        path: ""
      homepage_url: ""
      scope_names: []
    packages: []
    dependency_graphs:
      Gradle:
        nodes: []
        edges: []
scanner: null
advisor: null
evaluator: null
resolved_configuration:
  package_curations:
  - provider:
      id: "RepositoryConfiguration"
    curations: []
  - provider:
      id: "DefaultDir"
    curations: []
  - provider:
      id: "DefaultFile"
    curations: []

```

build.gradle.kts example (about 600 libraries)
```
repositories.maven("https://nexus.intra.example.com/repository/public")
repositories.maven("file:/home/user/.m2/repository/")
plugins.apply("java")
configurations.all {
    isTransitive = false
}
dependencies.add("runtimeOnly", "antlr:antlr:2.7.7")
dependencies.add("runtimeOnly", "ch.qos.logback.db:logback-core-db:1.2.11.1")
...
```

### Environment

* ORT version: 31.1.0
* Java version: 17
* OS: Linux
* ScanCode version: 32.2.1
* 8 CPUs and a maximum of 19664 MiB of memo

*And* specify (relevant parts of) your ORT configuration (`config.yml`):

```
ort:
  forceOverwrite: false
  addAuthorsToCopyrights: true
  enableRepositoryPackageCurations: true 
  enableRepositoryPackageConfigurations: true

  licenseFilePatterns:
    licenseFilenames: [ 'license*' ]
    patentFilenames: [ patents ]
    rootLicenseFilenames: [ 'readme*' ]

  severeIssueThreshold: ERROR
  severeRuleViolationThreshold: ERROR

  analyzer:
    allowDynamicVersions: true
    enabledPackageManagers: [ Gradle, Yarn, NPM ]

  downloader:
    allowMovingRevisions: true
    # Only used if the '--license-classifications-file' option is specified.
    includedLicenseCategories:
      - copyleft
      - copyleft-provide-sourcecode
      - weak-copyleft
      - weak-copyleft-provide-sourcecode
      - proprietary
      - permissive
      - public-domain
      - no-assertion
      - not-for-commercial-use
      - include-in-notice-file
      - include-source-code-offer-in-notice-file

  scanner:
    skipConcluded: true

    archive:

      postgresStorage:
        connection:
          url: ${POSTGRES_URL}
          schema: public
          username: ort
          password: ${POSTGRES_PASSWORD}
          sslmode: disable

    createMissingArchives: true

    storages:
      clearlyDefined:
        serverUrl: 'https://api.clearlydefined.io'

      postgres:
        connection:
          url: ${POSTGRES_URL}
          schema: public
          username: ort
          password: ${POSTGRES_PASSWORD}
          sslmode: disable

    storageReaders: [ postgres, clearlyDefined ]
    storageWriters: [ postgres ]

    provenanceStorage:

      postgresStorage:
        connection:
          url: ${POSTGRES_URL}
          schema: public
          username: ort
          password: ${POSTGRES_PASSWORD}
          sslmode: disable```

### Additional context

cf. Description above

**Comment by sschuberth** [comment]
> `enabledPackageManagers: [ Gradle, Yarn, NPM ]`

Please use "GradleInspector" instead of "Gradle" and give it a try again.

**Comment by schvvarzekatze** [comment]
Thank you for this info. With this config it seems to analyze with GradleInspector  `build.gradle.kts`but crashes ifout any `analyzer-result.yml` being generated.

```
/        \_______   \__    ___/ The OSS Review Toolkit, version DOCKER-SNAPSHOT
|    |   | |       _/ |    |    built with JDK 11.0.24+8, running under Java 17
|    |   | |    |   \ |    |    Executing 'analyze' as 'root' on Linux         
\________/ |____|___/ |____|    with 8 CPUs and a maximum of 19664 MiB of memor
                                                                               
Environment variables:                                                        
ORT_CONFIG_DIR = /xyz-project/ort                                             
ORT_DATA_DIR = /root/.ort                                                     
HOME = /home/ort                                                              
JAVA_HOME = /opt/java/openjdk                                                 
ANDROID_HOME = /opt/android-sdk                                               
                                                                              
Looking for ORT configuration in the following file:
        /xyz-project/ort/config.yml

Looking for analyzer-specific configuration in the following files and directories:
        /project/.ort.yml
        /xyz-project/ort/resolutions.yml
The following 3 package manager(s) are enabled:
        GradleInspector, Yarn, NPM
The following 3 package curation provider(s) are enabled:
        RepositoryConfiguration, DefaultDir, DefaultFile
Analyzing project path:
        /project
07:17:00.572 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/project/src/main/resources' as it is hard-coded to be ignored.
Found 1 GradleInspector definition file(s) at:
        build.gradle.kts
Found in total 1 definition file(s) from the following 1 package manager(s):
        GradleInspector
07:17:00.637 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManagerRunner - Starting GradleInspector analysis.
07:17:00.641 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Using GradleInspector to resolve dependencies for path 'build.gradle.kts'...
Downloading https://services.gradle.org/distributions/gradle-8.10-bin.zip
..................................................................................................................................
07:17:21.904 [DefaultDispatcher-worker-1] INFO  org.gradle.tooling.GradleConnector - The project at '/project' uses Gradle version 8.10.0.
07:18:04.721 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.ort.OrtAuthenticator - Authenticator was successfully installed.
07:18:04.738 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.ort.OrtProxySelector - Proxy selector was successfully installed.
07:18:09.724 [DefaultDispatcher-worker-1] INFO  OrtDependency - Maven SCM connection 'git@github.com:java-json-tools/btf.git' in 'com.github.java-json-tools:btf:1.3' lacks the required 'scm' prefix.
07:18:09.793 [DefaultDispatcher-worker-1] INFO  OrtDependency - Maven SCM connection 'git@github.com:java-json-tools/jackson-coreutils.git' in 'com.github.java-json-tools:jackson-coreutils:2.0' lacks the required 'scm' prefix.
07:18:10.035 [DefaultDispatcher-worker-1] INFO  OrtDependency - Maven SCM connection 'git@github.com:java-json-tools/json-patch.git' in 'com.github.java-json-tools:json-patch:1.13' lacks the required 'scm' prefix.
07:18:10.106 [DefaultDispatcher-worker-1] INFO  OrtDependency - Maven SCM connection 'git@github.com:java-json-tools/msg-simple.git' in 'com.github.java-json-tools:msg-simple:1.2' lacks the required 'scm' prefix.
07:18:13.742 [DefaultDispatcher-worker-1] INFO  OrtDependency - Maven SCM connection 'git@github.com:prowide/prowide-core.git' in 'com.prowidesoftware:pw-swift-core:SRU2021-9.2.8' lacks the required 'scm' prefix.
07:18:21.204 [DefaultDispatcher-worker-1] INFO  OrtDependency - Ignoring Maven SCM connection 'http://fisheye2.cenqua.com/browse/stringtemplate' in 'org.antlr:stringtemplate:3.2.1' due to an unexpected format.
07:18:34.692 [DefaultDispatcher-worker-1] INFO  OrtDependency - Maven SCM connection 'https://github.com/swagger-api/swagger-ui.git' in 'org.webjars:swagger-ui:5.17.14' lacks the required 'scm' prefix.
07:18:34.956 [DefaultDispatcher-worker-1] INFO  OrtDependency - Ignoring Maven SCM connection 'https://svn.apache.org/repos/asf/xerces/java/tags/Xerces-J_2_12_2/' in 'xerces:xercesImpl:2.12.2' due to an unexpected format.
07:18:35.272 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'hg root' in '/project'...
07:18:35.611 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.PackageManager - GradleInspector resolved dependencies for path 'build.gradle.kts' in 1m 34.965760927s.
07:18:35.625 [DefaultDispatcher-worker-1] INFO  kotlinx.coroutines.CoroutineScope - Finished GradleInspector analysis.
Exception in thread "main" java.lang.IllegalArgumentException: No DependencyGraph for package manager 'Gradle' available.
	at org.ossreviewtoolkit.model.DependencyGraphNavigator.graphForManager(DependencyGraphNavigator.kt:97)
	at org.ossreviewtoolkit.model.DependencyGraphNavigator.directDependencies(DependencyGraphNavigator.kt:48)
	at org.ossreviewtoolkit.analyzer.AnalyzerResultBuilderKt.resolvePackageManagerDependencies(AnalyzerResultBuilder.kt:117)
	at org.ossreviewtoolkit.analyzer.AnalyzerResultBuilderKt.access$resolvePackageManagerDependencies(AnalyzerResultBuilder.kt:1)
	at org.ossreviewtoolkit.analyzer.AnalyzerResultBuilder.build(AnalyzerResultBuilder.kt:51)
	at org.ossreviewtoolkit.analyzer.AnalyzerState.buildResult(Analyzer.kt:257)
	at org.ossreviewtoolkit.analyzer.Analyzer.analyzeInParallel(Analyzer.kt:182)
	at org.ossreviewtoolkit.analyzer.Analyzer.analyze(Analyzer.kt:134)
	at org.ossreviewtoolkit.plugins.commands.analyzer.AnalyzerCommand.run(AnalyzerCommand.kt:203)
	at com.github.ajalt.clikt.parsers.Parser.finalizeAndRun(Parser.kt:348)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:218)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:245)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:42)
	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:457)
	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:454)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:474)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:481)
	at org.ossreviewtoolkit.cli.OrtMainKt.main(OrtMain.kt:85)

```

**Comment by sschuberth** [comment]
Which version / commit of ORT are you using exactly?

**Comment by schvvarzekatze** [comment]
Sorry, I have directly used the version 31.1.0 which contains the commit [a477ded
](https://github.com/oss-review-toolkit/ort/commit/a477dedd42f7b65e740c54e018044bd4b62a6503)

**Comment by sschuberth** [comment]
> I can also use a snapshot of the newest version if this could help.

Yes please, there have been several GradleInspector-related fixes since version 31.1.0.

**Comment by schvvarzekatze** [comment]
Thank you it seems to work for version 33.1.0 in all steps for Gradle. For npm it also looks good for analyzer but not yet for the scan. 

So I will retry it with version 34.0.0 and close this issue referring to analyzer step.
